node/src/reducer.rs

```diff
@@ -17,7 +17,32 @@ pub fn reducer(
     let meta = action.meta().clone();
     match action.action() {
         Action::CheckTimeouts(_) => {
-            if state.p2p.ready().is_some() {
+            if let Some(p2p) = state.p2p.ready() {
+                if let Some(kad_state) = &p2p.network.scheduler.discovery_state {
+                    if p2p.ready_peers().is_empty() && kad_state.has_bootstraped {
```

Comment:
You should almost never had this much nesting in code. Please flatten out at least some of these conditionals:
```suggestion
            let is_bootstrapped_and_empty = state
                .p2p
                .ready()
                .map(|state| {
                    let has_bootstrapped = state
                        .netwowrk
                        .scheduler
                        .discovery_state
                        .map(|s| s.has_bootstraped)
                        .unwrap_or_default();
                    state.ready_peers().is_empty() && has_bootstrapped
                })
                .unwrap_or_default();
                
            if is_bootstrapped_and_empty {
```